alerting/src/main/kotlin/org/opensearch/alerting/util/destinationmigration/DestinationMigrationUtilService.kt

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util.destinationmigration
+
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.bulk.BulkRequest
+import org.opensearch.action.bulk.BulkResponse
+import org.opensearch.action.delete.DeleteRequest
+import org.opensearch.action.search.SearchRequest
+import org.opensearch.action.search.SearchResponse
+import org.opensearch.action.support.WriteRequest
+import org.opensearch.alerting.core.model.ScheduledJob
+import org.opensearch.alerting.model.destination.Destination
+import org.opensearch.alerting.model.destination.email.EmailAccount
+import org.opensearch.alerting.model.destination.email.EmailGroup
+import org.opensearch.alerting.opensearchapi.suspendUntil
+import org.opensearch.alerting.util.destinationmigration.DestinationConversionUtils.Companion.convertDestinationToNotificationConfig
+import org.opensearch.alerting.util.destinationmigration.DestinationConversionUtils.Companion.convertEmailAccountToNotificationConfig
+import org.opensearch.alerting.util.destinationmigration.DestinationConversionUtils.Companion.convertEmailGroupToNotificationConfig
+import org.opensearch.alerting.util.destinationmigration.NotificationApiUtils.Companion.createNotificationConfig
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.Strings
+import org.opensearch.common.xcontent.LoggingDeprecationHandler
+import org.opensearch.common.xcontent.NamedXContentRegistry
+import org.opensearch.common.xcontent.XContentFactory
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.common.xcontent.XContentType
+import org.opensearch.commons.ConfigConstants
+import org.opensearch.commons.notifications.action.CreateNotificationConfigRequest
+import org.opensearch.commons.notifications.model.NotificationConfig
+import org.opensearch.commons.notifications.model.NotificationConfigInfo
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.rest.RestStatus
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.opensearch.search.fetch.subphase.FetchSourceContext
+import java.time.Instant
+
+class DestinationMigrationUtilService {
+
+    companion object {
+
+        private val logger = LogManager.getLogger(DestinationMigrationUtilService::class)
+
+        @Volatile
+        private var runningLock = false // In case 2 migrateDestinations() processes are running
+
+        // Used in DestinationMigrationCoordinator to cancel scheduled process
+        @Volatile
+        var finishFlag = false
```

Comment:
Did we want to make this flag a cluster setting to get reset later on?

Reply:
Let's treat that as a potential enhancement for now. If we see a need for it, we can follow-up to add it.